internal/repository/contact.go

```diff
@@ -54,16 +55,19 @@ func (c contact) Update(contact model.Contact) (model.Contact, error) {
 	return contact, nil
 }
 
-func (c contact) DeleteByID(id uint) error {
-	if _, err := c.GetByID(id); err != nil {
-		return err
-	}
+// Wykonujemy usunięcie nawet jak nie ma rekordu
+func (c contact) DeleteByID(userID, id uint) error {
+
+	result := c.db.Where("id = ? AND user_id = ?", id, userID).Delete(&model.Contact{})
 
-	result := c.db.Delete(&model.Contact{}, id)
 	if result.Error != nil {
 		return result.Error
 	}
 
+	if result.RowsAffected == 0 {
+		return errors.New("no contact found or unauthorized")
```

Comment:
This error should not appear in repository, as this is business logic error. Return `(int64, error)` from function to check affected rows count in service